### ISSUE TITLE
CORS issue - can't view images from AWS through gif tool

### ISSUE THREAD

**Issue by apalmer0** [ga-wdi-boston/capstone-project#139]
I've got a list of URLs that point to images saved in AWS. When I try to use those images in my gif tool - which accepts URLs as parameters - I get an error telling me that doing so isn't allowed because of CORS restrictions. Specifically:

Image from origin 'https://s3.amazonaws.com' has been blocked from loading by Cross-Origin Resource Sharing policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:9000' is therefore not allowed access.

Any ideas?

**Comment by gaand** [comment]
I haven't had a chance to look at in detail, but this seems a likely place to find an answer:

http://docs.aws.amazon.com/AmazonS3/latest/dev/cors.html

**Comment by apalmer0** [comment]
Looked like it was a combo of that and [this stackoverflow answer](http://stackoverflow.com/questions/35002392/what-do-i-set-cors-settings-to-and-where-so-that-my-canvas-doesnt-get-tainted)

Before loading my images I had to set the crossorigin attribute to this:
`img.crossOrigin = "Anonymous";`

Which I tried before, but apparently the old, safe image was cached - so I had to clear the cache for it to work. Tricky. 

Ps I don't really get this so I'd love an explanation if someone gets a few spare minutes... I feel like I'm circumventing security measures that someone way smarter than me put in place.
